src/net.h

```diff
@@ -251,7 +252,12 @@ class CConnman
     void SetServices(const CService &addr, ServiceFlags nServices);
     void MarkAddressGood(const CAddress& addr);
     void AddNewAddresses(const std::vector<CAddress>& vAddr, const CAddress& addrFrom, int64_t nTimePenalty = 0);
-    std::vector<CAddress> GetAddresses();
+
+    // Cache is used to minimize topology leaks, so it should
+    // be used for all non-trusted calls, for example, p2p.
+    // A non-malicious call (from RPC or a whitelisted node)
+    // should avoid using the cache by passing nullopt.
```

Comment:
Can you elaborate on why a whitelisted node should avoid using the cache? If there are no real downsides to caching for 24 hours then this shouldn't matter, right?

Reply:
Some hypothetical examples which *might* make sense in a trusted setting:
- 2 remote nodes exchanging each others AddrMan and compare to make sure no poisoning is happening
- one node learns full AddrMan (or more than 1 cached list) from its *trusted* peer (doesn't really make sense in a public trust-minimized setting, but might make sense here)

I don't have a strong justification for this feature, but I thought it might be useful. Interested in other opinions!

Reply:
Yeah I don't see any downsides to maintaining the current behaviour for trusted callers. I was just confused at seeing it in the implementation but no mention of it in the PR description or discussion.